File: 07292021/powerOf2.cpp
// Print power of 2s inside a given number

#include <iostream>

using namespace std;

int powerOf2(int num)
{
	if (num < 0)
		return 0;
	else if (num == 1)
	{
		cout << 1 << ' ';
		return 1;
	}
	else
	{
		int prev = powerOf2(num / 2);
		int curr = prev * 2;
		cout << curr << ' ';
		return curr;
	}
}

int main()
{
	powerOf2(10);
	return 0;
}

File: 07292021/powerOf2.js
// Print all the power of 2 container in the number

const powerOf2 = (num) => {
    if (num < 1) return 0;
    if (num === 1) {
        console.log(1);
        return 1;
    }
    let prev = powerOf2(Math.floor(num / 2));
    let curr = prev * 2;
    console.log(curr);
    return curr;
};

powerOf2(50);
